frontend/catalyst/jax_tracer.py

```diff
@@ -453,7 +454,12 @@ def _bind_native_controlled_op(qrp, op, controlled_wires, controlled_values):
         return qrp
 
     qrp = QRegPromise(qreg)
-    for op in device.expand_fn(quantum_tape):
+    if isinstance(device, qml.Device):
+        ops = device.expand_fn(quantum_tape)
+    else:
+        ops = quantum_tape
```

Comment:
It looks we do the old tape expansion here in `trace_quantum_tape` and the new pre-processing with all other transforms in `trace_quantum_functions`. For code organization purposes, is it possible to have them in the same function (whether here or there, likely there I imagine - e.g. wrap the expand_fn call into an element of the TransformProgram) so that they are easy to identify as alternative pipelines, or would that break things?

Reply:
It looks like lightning.qubit will, be kokkos and gpu will stay on the old one for longer

Reply:
Okay so the requirement for backwards compatibility will be here for the near feature.